stdlib/public/core/CharacterProperties.swift

```diff
@@ -0,0 +1,293 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2018 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+extension Character {
+  @inlinable
+  internal var _firstScalar: Unicode.Scalar {
+    return self.unicodeScalars.first!
+  }
+  @inlinable
+  internal var _isSingleScalar: Bool {
+    return self.unicodeScalars.count == 1
+  }
+
+  @inlinable
+  static internal var _crlf: Character { return "\r\n" }
+
+  @inlinable
+  static internal var _lf: Character { return "\n" }
```

Comment:
The `_lf` property isn't used in this pull request.

The `_crlf` property is only used once.